Source/WebKit/UIProcess/PeriodicalResponsivenessTimer.h

```diff
@@ -0,0 +1,53 @@
+/*
+ * Copyright (C) 2022 Igalia S.L.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions
+ * are met:
+ * 1. Redistributions of source code must retain the above copyright
+ *    notice, this list of conditions and the following disclaimer.
+ * 2. Redistributions in binary form must reproduce the above copyright
+ *    notice, this list of conditions and the following disclaimer in the
+ *    documentation and/or other materials provided with the distribution.
+ *
+ * THIS SOFTWARE IS PROVIDED BY APPLE INC. AND ITS CONTRIBUTORS ``AS IS''
+ * AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO,
+ * THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR
+ * PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL APPLE INC. OR ITS CONTRIBUTORS
+ * BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+ * CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+ * SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+ * INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+ * CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ * ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF
+ * THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+#pragma once
+
+#include <wtf/RunLoop.h>
+
+namespace WebKit {
+
+class WebPageProxy;
+
+class PeriodicalResponsivenessTimer {
+    WTF_MAKE_FAST_ALLOCATED;
+public:
+    explicit PeriodicalResponsivenessTimer(WebPageProxy&);
+    ~PeriodicalResponsivenessTimer();
+
+    void start();
+    void stop();
```

Comment:
These are public but only used internally, I thought we were going to start/stop the timer depending on whether the view is currently visible. If I've understood the patch correctly we will wake up every 3 seconds for every web view just to trigger the responsiveness timer

Reply:
<< I thought we were going to start/stop the timer depending on whether the view is currently visible
Yes, but Miguel commented in https://bugs.webkit.org/show_bug.cgi?id=246330#c14 that "there's the possibility that the browser is playing video while the window is not visible, so that's a case we should handle as well." so that is why I started it on WebKitWebView constructor.
<< If I've understood the patch correctly we will wake up every 3 seconds for every web view just to trigger the responsiveness timer
You are right and it is a mistake, the interval should be 10sec. I am fixing it right now,

Reply:
How about run the timer only if the web view is (a) visible, or (b) playing audio? If you're concerned about invisible video, then it must be because of audio, right?

Reply:
Ah, I see that's exactly what you suggested below. Great.